File: Printpress.UI/Printpress_Angular/src/app/features/reports/models/cash-by-document-report.dto.ts
export interface CashByDocumentReportDto {
  dateFrom: string | null;
  dateTo: string | null;
  cashAccountId: string | null;
  totalIn: number;
  totalOut: number;
  documents: CashDocumentGroupDto[];
}

export interface CashDocumentGroupDto {
  referenceType: string | number | null;
  referenceId: string | null;
  referenceTypeName: string;
  transactionCount: number;
  totalIn: number;
  totalOut: number;
  net: number;
}

File: Printpress.Application/Reports/DTO/CashByDocumentReportDto.cs
using System.Text.Json.Serialization;
using Printpress.Domain;

namespace Printpress.Application;

public class CashByDocumentReportDto
{
    public DateTime? DateFrom { get; set; }
    public DateTime? DateTo { get; set; }
    public Guid? CashAccountId { get; set; }
    public decimal TotalIn { get; set; }
    public decimal TotalOut { get; set; }
    public List<CashDocumentGroupDto> Documents { get; set; } = [];
}

public class CashDocumentGroupDto
{
    [JsonConverter(typeof(JsonStringEnumConverter))]
    public CashTransactionReferenceType? ReferenceType { get; set; }

    public Guid? ReferenceId { get; set; }
    public string ReferenceTypeName { get; set; }
    public int TransactionCount { get; set; }
    public decimal TotalIn { get; set; }
    public decimal TotalOut { get; set; }
    public decimal Net { get; set; }
}
